src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -93,7 +93,11 @@ public async override Task OnDefinitionFoundAsync(DefinitionItem definition)
                 if (definitionItem.Location != null)
                 {
                     AddAndReportResultsIfAtMax(definitionItem);
-                    _id++;
+
+                    lock (_gate)
+                    {
+                        _id++;
+                    }
```

Comment:
1. it's not just the adding of ids.  two threads could call into OnDefinitionFoundAsync at the same time and read out the old value of _id and both share that id. 
2. _definitionToId is likely not threadsafe either.  so you can get corruption there.  

we likely need to chat about thread-safety.  how familiar are you with it?

Reply:
Could the issue with missing reported symbols perhaps have to do with some corruption here, where multiple symbols get reported with the same ID and so don't show up?  And running under the debugger "fixes" the race?

Reply:
quite possibly.  @allisonchou what are you visibly "seeing" as the wrong experience?  you say you're not getting all results.. but what does that mean in particular?

Reply:
If the batch size is 32 for example and there's 100 results, then the first 96 results are getting reported, but the last 4 are not.

Reply:
I altered the code in the latest commit so that anything involving IDs is locked, but the problem still seems to persist.